### ISSUE TITLE
Feature request: align match clauses with the match keyword

### ISSUE THREAD

**Issue by ceastlund** [ocaml-ppx/ocamlformat#1067]
We currently get this formatting when match is inside parentheses:
```
let _ =
  match () with
  | _ ->
    (match () with
    | _ -> ())
;;
```
We would prefer this formatting (indenting the final match clause one extra space, to line up with `match`), in large part because it is consistent with ocp-indent, and also because I think it makes sense to align clauses with the match they came from:
```
let _ =
  match () with
  | _ ->
    (match () with
     | _ -> ())
;;
```

This is with ocamlformat built from 99444f3054e29e2322e56b23422e9984a378170f, using `--profile=janestreet`.

**Comment by gpetiot** [comment]
I came across this old issue, if I remember @jberdine was opposing this formatting. Josh what were the reasons for that?
Besides these reasons it would mean either create a huge diff on existing formatted codebases, or add another option for this. (no ideal solution)

**Comment by jberdine** [comment]
At least originally, opening parens (that are just for disambiguation) were considered to be part of the expression, and so further indentation starts from the beginning of the paren. I think that was pretty uniformly applied, but now there are enough options that it is likely that it isn't consistent in some configurations (that's just the nature of configs). So in e.g.
```
( if b then
  true-arm
else
  false-arm )
```
the `true-arm` is also indented from the paren, just like the cases of the match are indented from the paren in the OP.

My position is that there is an at-least almost consistent style following this high-level point with the `ocamlformat` profile, and that should be preserved. There are other options to e.g. violate this for `if`-`then`-`else`, so there could be for `match`-`with` too. Maybe even the existing option could be extended/generalized to avoid adding more options unless strictly necessary. At a high level I think it would make sense to uniformly support the "indent from after opening parens" style as well as the existing "indent from before the opening parens" one.

**Comment by gpetiot** [comment]
Sorry for the late reply, @ceastlund is that something that you still wish for the `janestreet` profile? This would be quite an extensive change, basically branching everytime we open a parenthesis whether we open a formatting box before or after the `(`.

**Comment by ceastlund** [comment]
I think we would prefer it, but it is not high priority. It sounds like, at least at the moment, it is not worth the effort.

**Comment by gpetiot** [comment]
Fixed by #2102
